src/mixins/filing-template-mixin.ts

```diff
@@ -959,4 +975,29 @@ export default class FilingTemplateMixin extends DateMixin {
     }
     return toReturn
   }
+
+  /**
+   * Ensure consistent object structure for a Restoration, whether it contains a Name
+   * Request or not, and whether it is an initial draft or it has been previously saved.
+   * Object merging does not work very well otherwise due to nested properties.
+   * @param filing the filing fetched from legal-api
+   * @returns the filing in safe-empty state if applicable
+   */
+  formatEmptyRestoration (filing: any): RestorationFilingIF {
```

Comment:
Rather than `any`, can we declare a specific filing type?

Reply:
I can think of 2 better solutions for this issue:
1. when Filings UI create a draft restoration, it pre-populates these empty objects, or
2. create an "empty restoration" object that you can use in App.vue to fill in missing values.

Example of 2:
- create something like this: https://github.com/bcgov/business-create-ui/blob/cf242543f0aee56e94832628545f56f482a37a55/src/interfaces/common/address-interface.ts#L14
- use it in App.vue like this:
```
        draftFiling = {
          ...emptyRestorationFiling, // pre-fill needed properties         
          ...this.buildRestorationFiling(), // fill properties from store data
          ...draftFiling // fill properties from draft
        }
```

Reply:
The root problem is that we don't want the store to have invalid data, for example a null object when it should be an empty object. We should work to fix that instead of creating special cases.

Reply:
I'm happy with this conversation. @jonathan-longe Please resolve if OK with you, or follow up as needed.